### PR TITLE
Update binutils to new syntax and version 2.39

### PR DESCRIPTION
**Pull request by mofeing**
Current Binutils recipe not only does not longer work but also generated artifacts are wrong (it provides *.exe binaries in Linux...).

This PR updates to the syntax, exports `libbfd`, `libctf`, `libopcodes` shared libraries and refactors the recipe.

### CODE REVIEW

File: B/Binutils/build_tarballs.jl [hunk 1]
```diff
@@ -1,93 +1,84 @@
...
+    LibraryProduct("libbfd", :libbfd),
+    LibraryProduct("libctf", :libctf),
+    LibraryProduct("libopcodes", :libopcodes),
 ]
 
-# Dependencies that must be installed before this package can be built
-dependencies = [
-    "https://github.com/staticfloat/KernelHeadersBuilder/releases/download/v4.12.0-0/build_KernelHeaders.v4.12.0.jl",
-]
+dependencies = []
```
Comment (giordano) [inline B/Binutils/build_tarballs.jl:82]:
```suggestion
dependencies = Dependency[]
```

File: B/Binutils/build_tarballs.jl [hunk 2]
```diff
@@ -1,93 +1,84 @@
...
-script = """
+script = raw"""
 # FreeBSD build system for binutils apparently requires that uname sit in /usr/bin/
-ln -sf \$(which uname) /usr/bin/uname
-
-# On MacOS, we don't actually install Binutils.  We install libtapi and cctools.  Le sigh.
-if [[ $(compiler_target) == *apple* ]]; then
-    mkdir -p \${WORKSPACE}/srcdir/apple-libtapi/build
-    cd \${WORKSPACE}/srcdir/apple-libtapi/build
+ln -sf $(which uname) /usr/bin/uname
```
Comment (giordano) [inline B/Binutils/build_tarballs.jl:13]: Is this still needed?

File: B/Binutils/build_tarballs.jl [hunk 3]
```diff
@@ -1,93 +1,84 @@
 using BinaryBuilder
 
-compiler_target = triplet(platform_key(ARGS[end]))
-if compiler_target == "unknown-unknown-unknown"
-    error("This is not a typical build_tarballs.jl!  Must provide exactly one platform as the last argument!")
-end
-deleteat!(ARGS, length(ARGS))
+name = "Binutils"
+version = v"2.39"
```
Comment (giordano) [inline B/Binutils/build_tarballs.jl:4]:
```suggestion
version_string = "2.39"
version = VersionNumber(version_string)
```

File: B/Binutils/build_tarballs.jl [hunk 4]
```diff
@@ -1,93 +1,84 @@
...
-# Collection of sources required to build Binutils
 sources = [
-    "https://ftp.gnu.org/gnu/binutils/binutils-2.24.tar.bz2" =>
-	"e5e8c5be9664e7f7f96e0d09919110ab5ad597794f5b1809871177a0f0f14137",
-    "https://github.com/tpoechtrager/apple-libtapi.git" =>
-    "e56673694db395e25b31808b4fbb9a7005e6875f",
-    "https://github.com/tpoechtrager/cctools-port.git" =>
-    "ecb84d757b6f011543504967193375305ffa3b2f",
-    "./bundled",
+    ArchiveSource("https://ftp.gnu.org/gnu/binutils/binutils-2.39.tar.xz", "645c25f563b8adc0a81dbd6a41cffbf4d37083a382e02d5d3df4f65c09516d00"),
```
Comment (giordano) [inline B/Binutils/build_tarballs.jl:7]:
```suggestion
    ArchiveSource("https://ftp.gnu.org/gnu/binutils/binutils-$(version_string).tar.xz", "645c25f563b8adc0a81dbd6a41cffbf4d37083a382e02d5d3df4f65c09516d00"),
```

File: B/Binutils/build_tarballs.jl [hunk 5]
```diff
@@ -1,93 +1,84 @@
...
+            ln -s $fbase ${prefix}/bin/${fbase#${target}-}
+        fi
+    done
+fi
 """
 
-# These are the platforms we will build for by default, unless further
-# platforms are passed in on the command line
-platforms = [Platform("x86_64", "linux")]
+platforms = supported_platforms(; exclude=x -> os(x) != "linux")
```
Comment (giordano) [inline B/Binutils/build_tarballs.jl:55]: Can simplify to
```suggestion
platforms = supported_platforms(; exclude=!Sys.islinux)
```

**Review by giordano — COMMENTED** [review]
Wow, that was an old recipe

File: B/Binutils/build_tarballs.jl (hunk 2 above)
Comment (mofeing) [reply B/Binutils/build_tarballs.jl:13]: Not sure. I'm checking it in local.

File: B/Binutils/build_tarballs.jl [hunk 6]
```diff
@@ -1,93 +1,69 @@
...
-    ln -s \$fbase \${prefix}/bin/\${fbase#$(compiler_target)-}
-done
+make -j${nproc}
+make install
 """
 
-# These are the platforms we will build for by default, unless further
-# platforms are passed in on the command line
-platforms = [Platform("x86_64", "linux")]
+platforms = supported_platforms(; exclude=!Sys.islinux)
```
Comment (mofeing) [inline B/Binutils/build_tarballs.jl:40]:
```suggestion
platforms = supported_platforms(; experimental=true)
```

Comment (giordano) [reply B/Binutils/build_tarballs.jl]: `experimental=true` is no-op nowadays, can just remove it.

**Pull request merged**